lib/IRGen/IRGen.cpp

```diff
@@ -165,6 +165,12 @@ swift::getIRTargetOptions(const IRGenOptions &Opts, ASTContext &Ctx) {
   TargetOpts.FunctionSections = Opts.FunctionSections;
 
   auto *Clang = static_cast<ClangImporter *>(Ctx.getClangModuleLoader());
+
+  // WebAssembly doesn't support atomics yet, see https://bugs.swift.org/browse/SR-12097
+  if (Clang->getTargetInfo().getTriple().isOSBinFormatWasm()) {
```

Comment:
Remove the extraneous braces.  IIRC, we set the `ThreadModel` for cygwin/MinGW as well.  Can we please set the thread model at one site?

Reply:
As far as I'm aware this is a single place where `ThreadModel` is used in the whole Swift project, it's not set for Cygwin anywhere. The only somewhat related settings for Cygwin that I see are set in `createTargetMachine`, but that doesn't give access to `TargetOpts`:

```swift
  if (EffectiveTriple.isArch64Bit() && EffectiveTriple.isWindowsCygwinEnvironment())
    cmodel = CodeModel::Large;
```